maven-api-impl/src/main/java/org/apache/maven/internal/impl/resolver/MavenVersionScheme.java

```diff
@@ -19,20 +19,35 @@
 package org.apache.maven.internal.impl.resolver;
 
 import org.apache.maven.api.di.Named;
-import org.apache.maven.api.di.Provides;
 import org.apache.maven.api.di.Singleton;
 import org.eclipse.aether.util.version.GenericVersionScheme;
+import org.eclipse.aether.version.InvalidVersionSpecificationException;
+import org.eclipse.aether.version.Version;
+import org.eclipse.aether.version.VersionConstraint;
+import org.eclipse.aether.version.VersionRange;
 import org.eclipse.aether.version.VersionScheme;
 
 /**
  * Default version scheme provider: provides singleton {@link GenericVersionScheme} instance.
  */
 @Singleton
 @Named
-public final class DefaultVersionSchemeProvider {
+public class MavenVersionScheme implements VersionScheme {
 
-    @Provides
-    static VersionScheme getVersionScheme() {
-        return new GenericVersionScheme();
+    private final VersionScheme delegate = new GenericVersionScheme();
```

Comment:
@cstamas fwiw, it would be easier if `GenericVersionScheme` was not `final`...

Reply:
any other thing to make non final in resolver? Resolver was notorious about making all stuff final, and I did not want to change a lot there... I can do a sweeping change

Reply:
> any other thing to make non final in resolver? Resolver was notorious about making all stuff final, and I did not want to change a lot there... I can do a sweeping change

That's really the one that needed a workaround in my recent changes...